### PR TITLE
Dynamic channels 15

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/channels/views.py [hunk 1]
```diff
@@ -1231,7 +1359,7 @@ def get_form_class(self):
 
             if channel_type == Channel.TYPE_ANDROID:
                 return UpdateAndroidForm
-            elif channel_type == Channel.TYPE_NEXMO:
+            elif channel_type == 'NX':
```
Comment (rowanseymour) [inline temba/channels/views.py]: Any way to abstract this so Channel doesn't depend on a specific type?

File: temba/channels/models.py [hunk 2]
```diff
@@ -321,7 +315,7 @@ class Channel(TembaModel):
 
     TWIML_CHANNELS = [TYPE_TWILIO, TYPE_VERBOICE, TYPE_TWIML]
 
-    NCCO_CHANNELS = [TYPE_NEXMO]
+    NCCO_CHANNELS = ['NX']
```
[the remark this answers is not in the document]
Comment (norkans7) [reply temba/channels/models.py]: I'll try this using an enum like the category field

File: temba/channels/models.py [hunk 3]
```diff
@@ -869,7 +807,7 @@ def get_ivr_client(self):
             return self.get_twiml_client()
         elif self.channel_type == Channel.TYPE_VERBOICE:  # pragma: no cover
             return self.org.get_verboice_client()
-        elif self.channel_type == Channel.TYPE_NEXMO:
+        elif self.channel_type == 'NX':
```
Comment (rowanseymour) [inline temba/channels/models.py]: I think the clients should live in the channel type packages, and then `ChannelType` would have a method like `get_ivr_client(self, org)`, but maybe best to do that in a different PR since it requires changing a few channel types, and perhaps gets complicated where types share clients. Sound good?

File: temba/channels/types/nexmo/views.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+import phonenumbers
+from django import forms
+from django.core.exceptions import ValidationError
+from django.http import HttpResponseRedirect
+from django.urls import reverse
+from django.utils.translation import ugettext_lazy as _
+from smartmin.views import SmartFormView
+
+from temba.channels.models import Channel
+from temba.channels.views import BaseClaimNumberMixin, ClaimViewMixin, NEXMO_SUPPORTED_COUNTRIES, \
```
Comment (rowanseymour) [inline temba/channels/types/nexmo/views.py]: Can either of these be moved into the nexmo channel package?

File: temba/channels/models.py (hunk 3 above)
Comment (norkans7) [reply temba/channels/models.py]: 👍

File: temba/channels/types/nexmo/views.py (hunk 4 above)
Comment (norkans7) [reply temba/channels/types/nexmo/views.py]: right

Comment (norkans7) [reply temba/channels/types/nexmo/views.py]: should we add a search_view attribute to move that the types packages too?

Comment (rowanseymour) [reply temba/channels/types/nexmo/views.py]: Hmm not sure, number searching seems pretty specific to twilio and nexmo, right? Whilst those things still live on org, its ok to have some stuff outside of the type packages. But maybe start thinking how we get those things out of org in the long term.

Comment (norkans7) [reply temba/channels/types/nexmo/views.py]: they are currently in the channels app and they the one using primary those lists of countries you mentioned above
